gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -0,0 +1,283 @@
+          'Per alternate INFO field {} does not have same cardinality as '
+          ' number of alternates: {} vs {}'.format(
+              field_name, len(variant_info_data),
+              len(proc_var._alternate_datas)))
+    for alt_index, info in enumerate(variant_info_data):
+      proc_var._alternate_datas[alt_index]._infos[field_name] = info
+
+  def _add_annotation(self, proc_var, field_name, data):
+    # type: (ProcessedVariant, str, List[str]) -> None
+    """Adds an annotation INFO field based on the format in the header.
+
+    Args:
+      proc_var (ProcessedVariant): The object to which the annotations are being
+        added.
+      field_name (str): The name of the annotation field.
+      data(List[str]): The data part of the field separated on comma.
+    """
+    if not self._header_fields.infos.has_key(field_name):
+      raise ValueError('{} INFO not found in the header, variant: {}'.format(
+          field_name, proc_var))
+    header_desc = self._header_fields.infos[field_name].desc
+    annotation_names = extract_annotation_names(header_desc)
+    for annotation_str in data:
+      annotations = _extract_annotation_list_with_alt(annotation_str)
+      if len(annotation_names) != len(annotations) - 1:
+        raise ValueError('Expected {} annotations, got {}'.format(
+            len(annotation_names), len(annotations) - 1))
+      found_alt = False
+      # This assumes that number of alternate bases and annotation segments
+      # are not too big. If this assumption is not true, we should replace the
+      # following loop with a hash table search and avoid the quadratic time.
+      for alt in proc_var._alternate_datas:
+        if alt.alternate_bases == annotations[0]:
+          found_alt = True
+          self._add_annotation_for_alt(
+              alt, field_name, annotations[1:], annotation_names)
+          break
+      # TODO(bashir2): Currently we only check exact matches of alternate bases
+      # which is not enough. We should implement the whole standard for finding
+      # alternate bases for an annotation list.
+      if not found_alt:
+        logging.warning('Could not find matching alternate bases for {} in '
+                        'annotation filed {} with content {}'. format(
+            annotations[0], field_name, annotation_str))
+
+  def _add_annotation_for_alt(
+      self, alt, field_name, annotations, annotation_names):
+    # type: (AlternateBaseData, str, List[str], List[str]) -> None
+    if not alt._infos.has_key(field_name):
+      alt._infos[field_name] = []
+    annotation_dict = {}
+    for i in range(len(annotations)):
+      annotation_dict[annotation_names[i]] = annotations[i]
+    alt._infos[field_name].append(annotation_dict)
+
+
+def _extract_annotation_list_with_alt(annotation_str):
+  # type: (str) -> List[str]
+  """Extracts annotations from an annotation INFO field.
+
+  This works by dividing the ``annotation_str`` on '|'. The first element is
```

Comment:
Please provide a small example of what this string looks like and what the return value is like.